### PR TITLE
Tabnet

### PR DESCRIPTION
**Pull request by longtle**
If you are opening a PR for `Official Notebooks` under the [notebooks/official](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/official) folder, follow this mandatory checklist:
- [x ] Use the [notebook template](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb) as a starting point.
- [ x] Follow the style and grammar rules outlined in the above notebook template.
- [x ] Verify the notebook runs successfully in Colab since the automated tests cannot guarantee this even when it passes.
- [x ] Passes all the required automated checks. You can locally test for formatting and linting with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).
- [ x] You have consulted with a tech writer to see if tech writer review is necessary. If so, the notebook has been reviewed by a tech writer, and they have approved it.
- [ x] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/official/CODEOWNERS) file under the `Official Notebooks` section, pointing to the author or the author's team.
- [x ] The Jupyter notebook cleans up any artifacts it has created (datasets, ML models, endpoints, etc) so as not to eat up unnecessary resources.

If you are opening a PR for `Community Notebooks` under the [notebooks/community](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/notebooks/community) folder:
- [ ] This notebook has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/community/CODEOWNERS) file under the `Community Notebooks` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

If you are opening a PR for `Community Content` under the [community-content](https://github.com/GoogleCloudPlatform/vertex-ai-samples/tree/main/community-content) folder:
- [ ] Make sure your main `Content Directory Name` is descriptive, informative, and includes some of the key products and attributes of your content, so that it is differentiable from other content
- [ ] The main content directory has been added to the [CODEOWNERS](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/community-content/CODEOWNERS) file under the `Community Content` section, pointing to the author or the author's team.
- [ ] Passes all the required formatting and linting checks. You can locally test with these [instructions](https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/CONTRIBUTING.md#code-quality-checks).

### CODE REVIEW

**Comment by longtle** [comment]
Can someone take a look at my request since we need it ready for blog post?

File: notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb [hunk 1]
```diff
@@ -0,0 +1,620 @@
+{
```
Comment (ivanmkc) [inline notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Where is the training and serving code?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: As our offline discussion, the purpose of notebook is visualization the TabNet's mask, which is used to provide explanation.

Comment (helinwang) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>I feel it would be good to show users how to get aggregated_mask_values from TabNet trained from https://github.com/GoogleCloudPlatform/vertex-ai-samples/pull/726 in this notebook. Otherwise users don't know how to get aggregated_mask_values. Is there any feature gap on TabNet on pipeline preventing for getting aggregated_mask_values?</p>

Comment (helinwang) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Once above is done, we can make the TabNet on pipeline training notebook link to this one, so the user journey is more complete.</p>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: The aggregated_mask_values is part of output from estimator so there is no gap here. Not all users needs to use explanation. I have no objection to integrate two notebooks together.

Comment (helinwang) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: Thanks, Long. Either two separate notebooks or a single one look good to me. I slightly prefer two separate ones and make the training one link the explanation one. Due to note all users using TabNet training would need explanation. But no strong opinion.

Would it be possible for you to add an optional step for getting the aggregated_mask_values from the result of TabNet pipeline? If users don't want to run the TabNet training pipeline, we can share an existing output for visualization. Then we can figure out merge vs. keep two separate notebooks.

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: Sounds good. Thank you for your sugesstion.
Since we need the notebook ready to publish the blog, I suggest to keep the content as it now.
I'll add the optional section of getting aggregated mask in the next PR.

Comment (helinwang) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: lgtm

Comment (reznitskii) [inline notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>The Overview section should "Include a paragraph or two explaining what this example demonstrates, who should be interested in it, and what you need to know before you get started." (https://github.com/GoogleCloudPlatform/vertex-ai-samples/blob/main/notebooks/notebook_template.ipynb)</p><p> </p><p>For example, we could start by introducing TabNet:</p><p>"TabNet model training is the training pipeline for&nbsp;<a href="https://arxiv.org/abs/1908.07442" target="_blank">TabNet model architecture</a>. It supports classification and regression. TabNet uses sequential attention to choose which features to reason from at each decision step, enabling interpretability and more efficient learning because the learning capacity is used for the most salient features."</p><p> </p><p>Then add an overview of the tutorial:</p><p>"The goal of the tutorial is to provide a sample plotting tool to visualize the output of TabNet, which is helpful in explaining the algorithm."</p><p> </p><p>The Objective section should provide "high level bullets for the steps of performed in the notebook". Could you please add this?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (reznitskii) [inline notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Is it correct to say that both the training and the prediction data come from "Synthetic_2 (Syn2) data, described in Section 4.1 of the <a href="https://arxiv.org/pdf/1802.07814.pdf" target="_blank">Learning to Explain</a> paper"? gs://cloud-samples-data/ai-platform-unified/datasets/tabnet/tab_net_output/syn2 is the part of the data that is used for prediction?</p><p> </p><p>Could we also make the training data available on Cloud Storage?</p><p>What about the trained TabNet model, where is that stored?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

Comment (reznitskii) [inline notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Links are incorrect.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>I updated the links.</p>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Thank you for suggestion. Added.</p>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>I uploaded the training data, too.</p><p>We don't make the model available. As our discussion with Helin Wang above, we try to get this PR check-in. I'll have a follow-up PR so that the user can get the output by themselves.</p>

Comment (reznitskii) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Link address is: https://colab.research.google.com/github/GoogleCloudPlatform/vertex-ai-samples/blob/master/notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb.ipynb</p><p> </p><p>Should be https://colab.research.google.com/github/GoogleCloudPlatform/vertex-ai-samples/blob/master/notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb</p>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: Updated. Thanks.

Comment (reznitskii) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>I suggest the following wording:</p><p> </p><p>"Sample prediction data is stored on Google Cloud at gs://cloud-samples-data/ai-platform-unified/datasets/tabnet/tab_net_output/syn2. You can use your own set of prediction data, but you must ensure that the format of the prediction data is the same as the format of the training data."</p><p> </p><p>Information about the training set and the model are better suited for the Dataset section at the top of the notebook. </p>

File: notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb [hunk 2]
```diff
@@ -0,0 +1,625 @@
...
```
Comment (reznitskii) [inline notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>I suggest the following for the Dataset section:</p><p> </p><p>This tutorial uses Synthetic_2 (Syn2) data, described in Section 4.1 of the&nbsp;<a href="https://arxiv.org/pdf/1802.07814.pdf" target="_blank">Learning to Explain</a>&nbsp;paper. The input feature X is generated from a 10-dimensional standard Gaussian. The response variable Y is generated from feature X[3:6] only. The data has been split into training and prediction sets and has been uploaded to Google Cloud Storage:</p><ul><li>Training data: gs://cloud-samples-data/ai-platform-unified/datasets/tabnet/tab_net_input/syn2_train.csv.</li><li>Prediction data: gs://cloud-samples-data/ai-platform-unified/datasets/tabnet/tab_net_output/syn2</li></ul><p> </p><p>At this time, the TabNet model file is not publicly available.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/vertex-ai-samples/pull/732/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb (hunk 1 above)
Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Done. Thanks.</p>

File: notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb (hunk 2 above)
Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Done. Thanks.</p>

File: notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb (hunk 1 above)
Comment (reznitskii) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: <p>Are you able to open these links yourself or do they only work once the pull request is accepted? What do I need to do to actually go through the notebook steps?</p>

Comment (longtle) [reply notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb:1]: No, the link is only ready when the notebook is checked in.
You can use this link to try it: https://colab.sandbox.google.com/github/longtle/vertex-ai-samples/blob/tabnet/notebooks/official/tabnet/ai-explanations-tabnet-algorithm.ipynb

**Review by andrewferlitsch — APPROVED** [review]
review updates already done
